pkg/networkmanager/firewall.jsx

```diff
@@ -171,6 +193,136 @@ class AddServicesBody extends React.Component {
         });
     }
 
+    parse_services(content) {
+        if (!content) {
+            console.warn("Couldn't read /etc/services");
+            return [ ];
+        }
+
+        var lines = content.split('\n');
+        var ret = {};
+
+        for (var i = 0; i < lines.length; i++) {
```

Comment:
How about `content.split('\n').forEach(line =>`, so that you avoid the explicit `i` and multiple array access?